sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -16,16 +16,21 @@
 
 
 @pytest.mark.parametrize(
-    "strategy, expected",
+    "strategy, expected, sample_weight",
```

Comment:
The test will need to be updated with the change that I asked earlier.

Reply:
I updated the test cases that I previously pushed. Unfortunately, I'm doubting on their relevance. Would they be acceptable as is, or should I find more specific cases ?

Reply:
I would like to have 2 specific checks that check the real behaviour of passing weights:

- Check that passing None and an array of 1 will be equivalent.
- Check that passing some zero weight would be equivalent to ignoring the sample and check the bins in these conditions.

Reply:
Unfortunately, I fail to understand why, the case where `sample_weight=None` for `n_bins=[2, 3, 3, 3]` and the case where `sample_weight=[1, 1, 1, 1]` for `n_bins=[2, 3, 3, 3]` are not equivalent (in test method `test_fit_transform_n_bins_array`).

The behaviors of `np.percentile`and `_weighted_percentile` with `sample_weight = [1, 1, 1, 1]` are supposed to be equivalent, aren't they ?

Reply:
Indeed, I just checked:

```python
>>> import numpy as np
>>> from sklearn.utils.stats import _weighted_percentile
>>> data = np.random.randn(100)
>>> np.percentile(data, [25, 50, 75])
array([-0.76701739,  0.0604021 ,  0.79485777])
>>> [_weighted_percentile(data, np.ones_like(data), p) for p in [25, 50, 75]]
[-0.7855144180141034, 0.05197426163774039, 0.7586492302901591]
```

It seems that this problem has been known for a while but not yet fixed:

- #17370
- #6189

An old attempt to fix it: #17377

So for now, we can just comment out this case with a TODO comment with a link to #17370 for explain why this test is commented out.

Reply:
I commented out the faulty test case in https://github.com/scikit-learn/scikit-learn/pull/22048/commits/30edabcbd6efb87d38bfb47f004ca36a7cbea664 (l.139 - 151 in test_discretization.py).

Reply:
I commented out the faulty test case in https://github.com/scikit-learn/scikit-learn/pull/22048/commits/30edabcbd6efb87d38bfb47f004ca36a7cbea664 (l. 139-151)